pkg/netpol/connlist/exposure_analysis.go

```diff
@@ -0,0 +1,157 @@
+type exposedPeer struct {
+	peer Peer
+	*peerExposureData
+}
+
+func (ep *exposedPeer) ExposedPeer() Peer {
+	return ep.peer
+}
+
+func xgressExposureListToXgressExposureDataList(xgressExp []*xgressExposure) []XgressExposureData {
+	res := make([]XgressExposureData, len(xgressExp))
+	for i := range xgressExp {
+		res[i] = xgressExp[i]
+	}
+	return res
+}
+
+func (ep *exposedPeer) IsProtectedByIngressNetpols() bool {
+	return ep.isIngressProtected
+}
+
+func (ep *exposedPeer) IngressExposure() []XgressExposureData {
+	return xgressExposureListToXgressExposureDataList(ep.ingressExposure)
+}
+
+func (ep *exposedPeer) IsProtectedByEgressNetpols() bool {
+	return ep.isEgressProtected
+}
+
+func (ep *exposedPeer) EgressExposure() []XgressExposureData {
+	return xgressExposureListToXgressExposureDataList(ep.egressExposure)
+}
+
+// ----------------------------------------------------
+
+// gets an exposure map and builds ExposedPeer list while refining exposure connections
+// for each peer refines its exposure data as followed:
+// * if a peer is not protected by netpols on any direction (ingress/egress) :
+// the xgress lists are already empty so will have an empty xgressExp list (output will display not protected)
+// * if a peer is protected and exposed to entire cluster on all conns:
+// the list will include only this (eliminates other conns with specific namespaces)
+// * if a peer is exposed to entire cluster on ingress/egress on a specific connection:
+// eliminate exposure-data to specific potential namespaces on same direction and same connection
+func buildExposedPeerListFromExposureMap(exposuresMap exposureMap) []ExposedPeer {
+	res := make([]ExposedPeer, 0)
+	for p, expData := range exposuresMap {
+		ingExp := make([]*xgressExposure, 0)
+		egExp := make([]*xgressExposure, 0)
+		if expData.isIngressProtected {
+			ingExp = append(ingExp, loopAndRefineXgressData(expData.ingressExposure)...)
+		}
+		if expData.isEgressProtected {
+			egExp = append(egExp, loopAndRefineXgressData(expData.egressExposure)...)
+		}
+		// final peer's exposure data
+		expInfo := &exposedPeer{
+			peer: p,
+			peerExposureData: &peerExposureData{
+				isIngressProtected: expData.isIngressProtected,
+				isEgressProtected:  expData.isEgressProtected,
+				ingressExposure:    ingExp,
+				egressExposure:     egExp,
+			},
+		}
+		res = append(res, expInfo)
+	}
+	return res
+}
+
+func loopAndRefineXgressData(xgressData []*xgressExposure) []*xgressExposure {
+	res := make([]*xgressExposure, 0)
+	// helping var
+	var entireClusterConn *common.ConnectionSet
+	for _, singleConn := range xgressData {
+		//  exposed to entire cluster on all conns - result will include this one general exposureConn
+		if singleConn.exposedToEntireCluster && singleConn.potentialConn.AllConnections() {
+			res = nil // remove previous conns if exist
+			res = append(res, singleConn)
+			break
+		}
+		if singleConn.exposedToEntireCluster {
+			entireClusterConn = singleConn.potentialConn
```

Comment:
the `entireClusterConn` should be assigned only if it does not have previous value with larger connection set. It should actually store the maximal connectionSet for exposure to all cluster's namespaces.

Reply:
done